crates/daphne-server/src/roles/leader.rs

```diff
@@ -165,16 +140,30 @@ impl crate::App {
                 .map_err(|e| fatal_error!(err = ?e, "failed to construct content-type header"))?,
         );
 
-        if let Some(bearer_token) = req.sender_auth.and_then(|auth| auth.bearer_token) {
+        if let Some(auth) = self
+            .service_config
+            .leader_to_helper_tokens
+            .get(&url.host().unwrap().to_owned())
+        {
+            headers.insert(
+                HeaderName::from_static(http_headers::DAP_AUTH_TOKEN),
+                HeaderValue::from_str(auth.as_str()).map_err(
+                    |e| fatal_error!(err = ?e, "failed to construct authentication header"),
+                )?,
+            );
+        }
+        #[cfg(feature = "test-utils")]
+        if let Some(auth) = self
+            .bearer_tokens()
+            .get(daphne::DapSender::Leader, task_id)
+            .await
+            .map_err(|e| fatal_error!(err = ?e, "failed to get leader bearer token"))?
+        {
             headers.insert(
                 HeaderName::from_static(http_headers::DAP_AUTH_TOKEN),
```

Comment:
Hmm, if we set this header in the previous `if let Some(auth) = ...` then we'll overwrite it here. I don't think that's what we want to do.

Reply:
Question: are there any conditions under which we expect the peer_auth token to be the same as one of the per-task tokens? I suspect not, but perhaps you're thinking further ahead than I am.

If this is not expected, I think it would be best to treat it as a fatal error.